### PR TITLE
feat(AssetDetailView): Adding price history chart

### LINKED ISSUE #6490: Display price history chart
### Description

figma: https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=6455%3A122774

Screen Shot 2022-07-17 at 2 31 42 PM

### Acceptance Criteria

- [x] step 1: integrate a chart library (license needs to be compatible)
- [X] clicking on a token on the asset list should display a new section (should already be done by @Khushboo-dev-cpp )
- [X] clicking back on the section should go back to the asset list (should already be done by @Khushboo-dev-cpp )
- [x] should be possible to filter by last hour, last day, 7 days, 6 months, 1 year and all
- [ ] [Backend to provide historical data](https://github.com/status-im/status-desktop/issues/7260)
note: the chart for the balance is out of scope

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #6490

### What does the PR do
adds a dummy price chart in assets detail view

### Affected areas
Assets detail view

### Screenshot of functionality (including design for comparison)
https://user-images.githubusercontent.com/31625338/190172322-891dc552-84a8-44fb-afda-076a00896ddb.mov

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: ui/imports/shared/views/AssetsDetailView.qml [hunk 1]
```diff
@@ -18,8 +19,78 @@ Item {
...
+        //dummy data
+        property real stepSize: 1000
+        property real minStep: 12000
+        property real maxStep: 22000
+
+        property var graphTabsModel: [{text: "Price", enabled: true}, {text: "Balance", enabled: false}]
+        property var timeRangeTabsModel: [{text: "1H", enabled: true},
+                {text: "1D", enabled: true},{text: "7D", enabled: true},
+                {text: "1M", enabled: true}, {text: "6M", enabled: true},
+                {text: "1Y", enabled: true}, {text: "ALL", enabled: true}]
```
Comment (caybro) [inline ui/imports/shared/views/AssetsDetailView.qml:33]: Here you need to add the `qsTr()` for `text`s :)

File: ui/imports/shared/views/AssetsDetailView.qml [hunk 2]
```diff
@@ -18,8 +19,78 @@ Item {
...
+        }
+
+        function day(before = 0) {
+            var newDay = new Date(Date.now() - before * 24 * 60 * 60 * 1000).toString();
+            return newDay.slice(4,7) + " " + newDay.slice(8,10);
+        }
+
+        function month(before = 0) {
+            var newMonth = new Date(Date.now() - before * 24 * 60 * 60 * 1000).toString();
+            return newMonth.slice(4,7) + " '" + newMonth.slice(newMonth.indexOf("G")-3, newMonth.indexOf("G")-1);
```
Comment (caybro) [inline ui/imports/shared/views/AssetsDetailView.qml:69]: What is this "G"?

File: ui/imports/shared/views/AssetsDetailView.qml [hunk 3]
```diff
@@ -18,8 +19,78 @@ Item {
...
+
+        property var timeRange: [
+            {'1H': [minutes(60), minutes(55), minutes(50), minutes(45), minutes(40), minutes(35), minutes(30), minutes(25), minutes(20), minutes(15), minutes(10), minutes(5), minutes()]},
+            {'1D': [hour(24), hour(23), hour(22), hour(21), hour(20), hour(19), hour(18), hour(17), hour(16), hour(15), hour(14), hour(13),
+                    hour(12), hour(11), hour(10), hour(9), hour(8), hour(7), hour(6), hour(5), hour(4), hour(3), hour(2), hour(1), hour()]},
+            {'7D': [day(6), day(5), day(4), day(3), day(2), day(1), day()]},
+            {'1M': [day(30), day(28), day(26), day(24), day(22), day(20), day(18), day(16), day(14), day(12), day(10), day(8), day(6), day(4), day()]},
+            {'6M': [month(150), month(120), month(90), month(60), month(30), month()]},
+            {'1Y': [month(330), month(300), month(270), month(240), month(210), month(180), month(150), month(120), month(90), month(60), month(30), month()]},
+            {'ALL': ['2016', '2017', '2018', '2019', '2020', '2021', '2022']}
```
Comment (caybro) [inline ui/imports/shared/views/AssetsDetailView.qml:80]: Should "2022" be hardcoded like that? Shouldn't it go all the way to the "current" year?

File: ui/imports/shared/views/AssetsDetailView.qml [hunk 4]
```diff
@@ -54,12 +125,126 @@ Item {
...
+            chart.chartType: 'line'
+            chart.chartData: {
+                return {
+                    labels: d.timeRange[graphDetail.timeRangeTabBarIndex][graphDetail.selectedTimeRange],
+                    datasets: [{
+                            label: 'Price',
+                            xAxisId: 'x-axis-1',
+                            yAxisId: 'y-axis-1',
+                            backgroundColor: (Theme.palette.name === "dark") ? 'rgba(136, 176, 255, 0.2)' : 'rgba(67, 96, 223, 0.2)',
+                            borderColor: (Theme.palette.name === "dark") ? 'rgba(136, 176, 255, 1)' : 'rgba(67, 96, 223, 1)',
```
Comment (caybro) [inline ui/imports/shared/views/AssetsDetailView.qml:158]: Should probably use colors coming from the Theme

File: ui/imports/shared/views/AssetsDetailView.qml [hunk 5]
```diff
@@ -54,12 +125,126 @@ Item {
...
+                d.generateData();
+                //if graph bar
+                //switch graph
+            }
+            chart.chartType: 'line'
+            chart.chartData: {
+                return {
+                    labels: d.timeRange[graphDetail.timeRangeTabBarIndex][graphDetail.selectedTimeRange],
+                    datasets: [{
+                            label: 'Price',
```
Comment (caybro) [inline ui/imports/shared/views/AssetsDetailView.qml:154]: `qsTr()`

File: ui/imports/shared/views/AssetsDetailView.qml [hunk 6]
```diff
@@ -54,12 +125,126 @@ Item {
...
+                                id: 'x-axis-1',
+                                position: 'bottom',
+                                gridLines: {
+                                    drawOnChartArea: false,
+                                    drawBorder: false,
+                                    drawTicks: false,
+                                },
+                                ticks: {
+                                    fontSize: 10,
+                                    fontColor: (Theme.palette.name === "dark") ? '#909090' : '#939BA1',
```
Comment (caybro) [inline ui/imports/shared/views/AssetsDetailView.qml:207]: Theme colors

File: ui/imports/shared/views/AssetsDetailView.qml [hunk 7]
```diff
@@ -54,12 +125,126 @@ Item {
...
+                                    padding: 24,
+                                    min: d.minStep,
+                                    max: d.maxStep,
+                                    stepSize: d.stepSize,
+                                    callback: function(value, index, ticks) {
+                                        return '$' + value;
```
Comment (caybro) [inline ui/imports/shared/views/AssetsDetailView.qml:232]: Hardcoded "$"?

**Review by caybro — CHANGES_REQUESTED** [review]

File: ui/imports/shared/views/AssetsDetailView.qml (hunk 3 above)
Comment (alexandraB99) [reply ui/imports/shared/views/AssetsDetailView.qml:80]: it's dummy data, I assume all these will be removed anyway when the backend part is ready, this is why i didn't put a lot of effort on this

File: ui/imports/shared/views/AssetsDetailView.qml (hunk 4 above)
Comment (alexandraB99) [reply ui/imports/shared/views/AssetsDetailView.qml:155]: yes, it didn't work. for some reason it behaved strangely so I had to set it like this

File: ui/imports/shared/views/AssetsDetailView.qml (hunk 7 above)
Comment (alexandraB99) [reply ui/imports/shared/views/AssetsDetailView.qml:229]: see above

File: ui/imports/shared/views/AssetsDetailView.qml (hunk 6 above)
Comment (alexandraB99) [reply ui/imports/shared/views/AssetsDetailView.qml:204]: see above

File: ui/imports/shared/views/AssetsDetailView.qml (hunk 1 above)
Comment (alexandraB99) [reply ui/imports/shared/views/AssetsDetailView.qml]: added

File: ui/imports/shared/views/AssetsDetailView.qml (hunk 5 above)
Comment (alexandraB99) [reply ui/imports/shared/views/AssetsDetailView.qml:151]: added

File: ui/imports/shared/views/AssetsDetailView.qml (hunk 4 above)
Comment (caybro) [reply ui/imports/shared/views/AssetsDetailView.qml:155]: Ah, it expects HTML rgba() colors probably

File: ui/imports/shared/views/AssetsDetailView.qml (hunk 3 above)
Comment (caybro) [reply ui/imports/shared/views/AssetsDetailView.qml:80]: Fair enough

**Review by caybro — APPROVED** [review]

**Pull request merged**